### PR TITLE
Binyamin/support descriptor extensions

### PR DESCRIPTION
**Pull request by binyamin555**

### CODE REVIEW

**Comment by binyamin555** [comment]
#6

File: src/extypes/__init__.py [hunk 1]
```diff
@@ -33,20 +33,32 @@ def extend_type_with(typ, cls):
...
+class ExtensionObject(Generic[_T]):
+    original: _T
+
+    def __init__(self, original: _T):
+        self.original = original
+
+
+def extension(o: _T) -> _T:
+    if callable(o) or hasattr(o, "__get__") or hasattr(o, "__set__") or hasattr(o, "__delete__"):
+        print("marked as extension:", o)
```
Comment (elazarcoh) [inline src/extypes/__init__.py:58]: don't forget to remove

**Review by elazarcoh — APPROVED** [review]
LGTM.
BTW we need to create a github actions for testing

File: src/extypes/__init__.py (hunk 1 above)
Comment (binyamin555) [reply src/extypes/__init__.py:58]: Right.. if one of you get to do that before that would also be ok, I got off for today

Comment (binyamin555) [reply src/extypes/__init__.py]: Done

**Review by neriyaco — CHANGES_REQUESTED** [review]
It looks like the `pytest` command is failing on the `test_length_function` function.
Also, add test to the descriptor extension under the tests directory.

**Review by neriyaco — APPROVED** [review]

**Comment by binyamin555** [comment]
> It looks like the `pytest` command is failing on the `test_length_function` function.
> Also, add test to the descriptor extension under the tests directory.

I changed it, however, I'm unable to test my changes due to platform incompatibility.

Can you review once again?

**Pull request merged**